apps/web/providers/PostHogProvider.tsx

```diff
@@ -0,0 +1,58 @@
+// app/providers.tsx
+"use client";
+import { useUser } from "@clerk/nextjs";
+import { usePathname, useSearchParams } from "next/navigation";
+import posthog from "posthog-js";
+import { PostHogProvider } from "posthog-js/react";
+import { useEffect } from "react";
+if (typeof window !== "undefined") {
+  posthog.init(process.env.NEXT_PUBLIC_POSTHOG_KEY!, {
+    api_host: process.env.NEXT_PUBLIC_POSTHOG_HOST!,
+    capture_pageview: false,
+    disable_session_recording: false,
+  });
+}
+
+export function PostHogPageview(): JSX.Element {
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
+  useEffect(() => {
+    if (pathname) {
+      let url = window.origin + pathname;
+      if (searchParams?.toString()) {
+        url = url + `?${searchParams.toString()}`;
+      }
+      posthog.capture("$pageview", {
+        $current_url: url,
+      });
+    }
+  }, [pathname, searchParams]);
+
+  return <></>;
```

Comment:
I think I read somewhere returning `null` is preferred here, but can't find it anymore

Reply:
This is from their docs

https://posthog.com/docs/libraries/next-js 

TS App Router